docs/messaging/coenrolling-features.mdx

```diff
@@ -0,0 +1,35 @@
+---
+id: coenrolling-features
+title: Co-enrolling Features
+slug: /messaging/coenrolling-features
+---
+
+A feature which allows co-enrollment allows a client to be enrolled in any number of experiments/rollouts for that feature. 
+
+**Features supporting co-enrollment**
+
+* Messaging (Fenix, Firefox iOS, Focus for Android, Focus for iOS)
+
+# How to define a co-enrolling feature
+
+A feature can be marked as allowing co-enrollment with a boolean flag in its feature definition in a Feature Manifest. For example, in your `messaging.fml.yaml` file:
+
+```yaml
+features:
+  messaging:
+    description: |
+      The in-app messaging system.
+    allow-coenrollment: true
+    variables:
+      messages:
+        description: A growable collection of messages
+        type: Map<String, MessageData>
+        default: {}
+```
+
+Once your messaging feature is defined as a co-enrolling feature, a client can be enrolled in any number of messaging experiments/rollouts.
+
+Other things to note: 
+* Experiment feature values still take precedence over rollout feature values
+* Enrollment/unenrollment is calculated independently for each experiment/rollout regardless of whether they target the same feature or not
+* Enrollment/unenrollment telemetry is still sent for each experiment/rollout at the time each enrolls or unenrolls
```

Comment:
Perhaps you can talk about recording exposure for coenrolling features here.

- The string `{experiment}` needs to exist in the feature config somewhere (e.g. in the case of messaging, a message object)
- It's replaced with the experiment slug at enrollment time.
- the feature code— that the developer is writing— needs to get the experiment slug from the feature config and record the exposure with `recordExperimentExposure(experimentSlug)`.